core/msg.c

```diff
@@ -370,6 +370,25 @@ static int _msg_receive(msg_t *m, int block)
     DEBUG("This should have never been reached!\n");
 }
 
+int msg_avail(void)
+{
+    unsigned state = disableIRQ();
+    DEBUG("msg_available: %" PRIkernel_pid ": msg_available.\n",
+          sched_active_thread->pid);
+
+    tcb_t *me = (tcb_t*) sched_threads[sched_active_pid];
```

Comment:
Yes, please, and in `msg_send_receive`, too. But that should go into a new commit. Could be that I am missing a use case where `sched_threads[sched_active_pid] != sched_active_thread`, but I think that these two need to match in all cases but for a short moment then switching threads.